vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/HasValidationProperties.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.shared;
+
+import com.vaadin.flow.component.HasElement;
+import com.vaadin.flow.component.HasValidation;
+
+/**
+ * Mixin interface for components that provide properties for setting invalid
+ * state and error message string to show when invalid.
+ *
+ * @author Vaadin Ltd
+ */
+public interface HasValidationProperties extends HasElement, HasValidation {
+
+    /**
+     * Sets the error message to show to the user when the component is invalid.
+     *
+     * @param errorMessage
+     *            the error message or {@code null} to clear it
+     */
+    @Override
+    default void setErrorMessage(String errorMessage) {
+        getElement().setProperty("errorMessage",
+                errorMessage == null ? "" : errorMessage);
+    }
+
+    /**
+     * Gets the error message to show to the when the component is invalid.
+     *
+     * @return the error message or {@code null} if not set
+     */
+    @Override
+    default String getErrorMessage() {
+        return getElement().getProperty("errorMessage", "");
```

Comment:
**note:** returning an empty string by default will be kind of a breaking change.

Reply:
Once we agree on the proper return type here, I believe BasicValidation unit tests can be removed in favor of simple `instanceof` checks.

Reply:
Restored the original behavior for now. We can consider changing this for all the properties later.